### PR TITLE
FIX Detect if the --column-statistics flag is available

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/sspak/issues/90

### CODE REVIEW

**Comment by LoveDuckie** [comment]
I'm encountering this problem as well and it has been affecting my backups. 😅 Hopefully a fix like this with conditional checks gets merged in soon.

**Comment by dhensby** [comment]
It seems to me that making sure if there are errors with the script that a non-zero exit code is returned is also important. Is that something that happens?

**Comment by LoveDuckie** [comment]
From what I experienced yes, but I can double check. I have a script that checks the exit code immediately after, and it would have triggered.

**Comment by dhensby** [comment]
Travis was automatically deploying built phars (https://github.com/silverstripe/sspak/blob/master/.travis.yml#L26-L27).

It looks like this may have stopped working - probably because the encypted GH token has expired. Moving this to GH actions shouldn't be any more complex than what is already in the travis config.

**Review by sabina-talipova — COMMENTED** [review]
@emteknetnz ,
I've done a few tests with using different DBs (MySQL 5.7.4 / 8.0.31, MariaDB 10.9.3). 
When I run `sspak save --db <from> <to>` I see message in terminal `mysqldump: unknown variable 'column-statistics=0'`, but it doesn't prevent me from creating DB dump file. After I use `sspak load` command it restore DB with content from dump file.
Since `mysqldump: unknown variable 'column-statistics=0'` is a result of mysqldump command execution, I think we should provide some sort of clarification in docs or in README.md, that user can see this message when they use `sspak save --db <from> <to>` on some DB.

File: README.md [hunk 1]
```diff
@@ -25,14 +25,21 @@ By convention, the file should have the extension `.sspak` for non-executable ve
...
 
 You can install this package globally with Composer (ensure your composer bin is in your system path):
 
     $> composer global require silverstripe/sspak:dev-master
 
+You can also require it directly in your project
+
+    $> composer require --dev silverstripe/sspak
+    $> vendor/bin/sspak <commands>
+
```
Comment (sabina-talipova) [inline README.md:38]: Could we mention that user might see a message `mysqldump: unknown variable 'column-statistics=0'`

**Review by sabina-talipova — CHANGES_REQUESTED** [review]

**Comment by emteknetnz** [comment]
> Moving this to GH actions shouldn't be any more complex than what is already in the travis config.

Honestly I'd be inclined to just drop support for the multiple ways of being able to install this thing and force the use of composer (I'm assuming that a global install works just as well as a project install).  I see no reason to have to maintain extra ways to install this, especially given the history of it silently breaking.

I've [backlogged](https://github.com/silverstripe/sspak/issues/92) a card for the team to look to see if there are any quick wins to salvage the old functionality - though my recommendation is to remove it.

**Comment by emteknetnz** [comment]
@sabina-talipova I've updated the readme

**Review by sabina-talipova — APPROVED** [review]

**Comment by OldStarchy** [comment]
I'm still getting this error with the latest version 548042a

```
mysqldump: unknown variable 'column-statistics=0'
```

My workaround was to manually replace these lines with `$columnStatistics = '';`

https://github.com/silverstripe/sspak/blob/548042abf5a890963f0240fc7c80acbe65044e9f/src/SSPak.php#L335-L342

Using MySQL 5.7.40-1.el7

**Comment by OldStarchy** [comment]
```bash
$ mysqldump --user=root --password=test --host=mysql site --column-statistics=0
mysqldump: unknown variable 'column-statistics=0'
$ echo $?
7
```

**Comment by sabina-talipova** [comment]
Hi @OldStarchy,
It works as was expected. In README we mention about this potential message, please see [Notes section here](https://github.com/creative-commoners/sspak/blob/6b3bb61572c117bd423b187ef1f4cf1f4a66e4bc/README.md#notes).
This is the result of `mysqldump` execution in this line https://github.com/silverstripe/sspak/blob/548042abf5a890963f0240fc7c80acbe65044e9f/src/SSPak.php#L335
It prevents further throwing an error:  https://github.com/silverstripe/sspak/blob/548042abf5a890963f0240fc7c80acbe65044e9f/src/SSPak.php#L344

**Pull request merged**